### PR TITLE
ocamlPackages.biniou: 1.2.1 → 1.2.2

### PR DESCRIPTION
**Pull request by vbgl**
###### Description of changes

Support for OCaml 5.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/biniou/default.nix [hunk 1]
```diff
@@ -1,29 +1,21 @@
...
 
-  useDune2 = true;
-
-  src = fetchFromGitHub {
-    owner = "ocaml-community";
-    repo = pname;
-    rev = version;
-    sha256 = "0x2kiy809n1j0yf32l7hj102y628jp5jdrkbi3z7ld8jq04h1790";
+  src = fetchurl {
+    url = "https://github.com/mjambon/biniou/releases/download/${version}/biniou-${version}.tbz";
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/biniou/default.nix:8]:
```suggestion
    url = "https://github.com/ocaml-community/biniou/releases/download/${version}/biniou-${version}.tbz"
```

https://github.com/ocaml-community/biniou/releases/download/1.2.2/biniou-1.2.2.tbz is available
Also general question on nixpkgs do we prefer GitHub tag or release ?

File: pkgs/development/ocaml-modules/biniou/default.nix [hunk 2]
```diff
@@ -1,29 +1,21 @@
-{ lib, fetchFromGitHub, buildDunePackage, easy-format }:
+{ lib, fetchurl, buildDunePackage, camlp-streams, easy-format }:
 
 buildDunePackage rec {
   pname = "biniou";
-  version = "1.2.1";
+  version = "1.2.2";
 
-  useDune2 = true;
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/biniou/default.nix:7]: duneVersion = "2";
?

**Review by Et7f3 — APPROVED** [review]
nixpkgs-review is happy on x86_64-darwin

File: pkgs/development/ocaml-modules/biniou/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/biniou/default.nix:7]: Rationale?

Comment (Et7f3) [reply pkgs/development/ocaml-modules/biniou/default.nix:7]: Easier to convert to dune3 with sed.

File: pkgs/development/ocaml-modules/biniou/default.nix [hunk 3]
```diff
@@ -1,29 +1,21 @@
...
-
-  src = fetchFromGitHub {
-    owner = "ocaml-community";
-    repo = pname;
-    rev = version;
-    sha256 = "0x2kiy809n1j0yf32l7hj102y628jp5jdrkbi3z7ld8jq04h1790";
+  src = fetchurl {
+    url = "https://github.com/ocaml-community/biniou/releases/download/${version}/biniou-${version}.tbz"
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/biniou/default.nix:8]:
```suggestion
    url = "https://github.com/ocaml-community/biniou/releases/download/${version}/biniou-${version}.tbz";
```

my bad

**Pull request merged**